smart-hadoop-support/smart-hadoop-3.2/src/main/java/org/smartdata/hdfs/compression/Lz4CompressorFactory.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.smartdata.hdfs.compression;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.compress.Compressor;
+import org.apache.hadoop.io.compress.Decompressor;
+import org.apache.hadoop.io.compress.Lz4Codec;
+import org.apache.hadoop.io.compress.lz4.Lz4Compressor;
+import org.apache.hadoop.io.compress.lz4.Lz4Decompressor;
+
+import java.io.IOException;
+
+public class Lz4CompressorFactory implements CompressorFactory, DecompressorFactory {
+  public static final String LZ4_CODEC = "Lz4";
+
+  @Override
+  public String codec() {
+    return LZ4_CODEC;
+  }
+
+  @Override
+  public int compressionOverhead(int bufferSize) {
+    return bufferSize / 255 + 16;
```

Comment:
please, write some comment about these magic numbers

Reply:
Previously, I added a TODO for further investigation into these magic numbers, as the sources of the Hadoop classes where they were copied from don't provide any documentation explaining the numbers. Additionally, I did not find any information on the internet during brief research.  https://github.com/arenadata/SSM/blob/ecdfedc2aaece2e718190afbd298a805a1a0ae99/smart-hadoop-support/smart-hadoop-common/src/main/java/org/smartdata/hdfs/compression/CompressorFactory.java#L35

Added more comments and copied TODO's to interface implementations